### PR TITLE
Retire Jekyll 2 docs

### LINKED ISSUE #5741: Remove requirements for Jekyll 2
http://jekyllrb.com/docs/installation/ still mentions Jekyll 2 dependencies — an old Ruby version, NodeJS and Python. I don't know how many people would like to install Jekyll 2 today, version 3 was published on October 27th 2015, maybe it's time to remove those instructions?

/cc @jekyll/core @jekyll/documentation

### PR DESCRIPTION
**Pull request by chrisfinazzo**
cc/ @jekyll/documentation.

### CODE REVIEW

**Comment by DirtyF** [comment]
fixes #5741

File: docs/_docs/upgrading.md [hunk 1]
```diff
@@ -6,5 +6,4 @@ permalink: /docs/upgrading/
 
 Upgrading from an older version of Jekyll? Upgrading to a new major version of Jekyll (e.g. from v2.x to v3.x) may cause some headaches. Take the following guides to aid your upgrade:
```
Comment (DirtyF) [inline docs/_docs/upgrading.md]: ➡️  This page could now redirect to  /upgrading/2-to-3/

Comment (chrisfinazzo) [inline docs/_docs/upgrading.md]: OK, will update. Am I correct in that this is just another entry inside of `docs/redirects`?

Comment (DirtyF) [inline docs/_docs/upgrading.md]: I think these are for external redirects.

File: docs/_docs/upgrading.md [hunk 2]
```diff
@@ -6,5 +6,4 @@ permalink: /docs/upgrading/
...
 
-- [From 0.x to 1.x and 2.x](/docs/upgrading/0-to-2/)
```
Comment (parkr) [inline docs/_docs/upgrading.md]: I'd keep this link!

File: docs/_docs/upgrading/0-to-2.md [hunk 3]
```diff
@@ -1,140 +0,0 @@
----
-layout: docs
-title: Upgrading from 0.x to 2.x
-permalink: /docs/upgrading/0-to-2/
----
-
-Upgrading from an older version of Jekyll? A few things have changed in 1.0
```
Comment (parkr) [inline docs/_docs/upgrading/0-to-2.md]: Let's leave this document – it's still useful. Believe it or not, there are plenty of folks using Jekyll 0.12.

Comment (DirtyF) [inline docs/_docs/upgrading/0-to-2.md]: I was wondering, is there any way to know when a version gets downloaded from Rubygems?

Comment (parkr) [inline docs/_docs/upgrading/0-to-2.md]: There aren't any hooks, but we could poll: http://guides.rubygems.org/rubygems-org-api/#gem-version-methods

Comment (pathawks) [inline docs/_docs/upgrading/0-to-2.md]: I worry that somebody might inherit a Jekyll site that hasn't been updated for years. Even if we aren't supporting 2.0, we should still provide a path for Jekyll 1.0 users to upgrade their sites to the latest and greatest.

Comment (DirtyF) [inline docs/_docs/upgrading/0-to-2.md]: I thought we could assume we provide documentation for latest version by default, but we can point to an older version like 2.0 ou 0.12 docs version.

**Comment by DirtyF** [comment]
FYI [A recent Twitter poll](https://twitter.com/jekyllrb/status/821152178492600321) show that 14% of our users are still using Jekyll v2 or less. That's quiet a lot, those people should be able to access a documentation adapted for the version they still use, even if we strongly invite them to update.

Shouldn't we build a documentation corresponding to latest v2 and hosting it on something like https://v2.jekyllrb.com/ ?

**Comment by chrisfinazzo** [comment]
I think (long-term) that's the direction we need to move in with regard to old versions that are no longer being targeted with new features, but still need support. At the same time, it seems the navigation needs to be rethought given the comments in #5698 - adding more items to the list is untenable.

Using a separate domain would make it clear about how to retire "active" documentation as new versions get released, so I'm 👍🏼 on this.

**Comment by parkr** [comment]
I think we should keep these around. We can do a better job of *isolating* the Jekyll 2 docs so as not to confuse users, but we shouldn't get rid of instructions when some folks are still using them.

**Pull request closed without merging**